File: 2025/7. Reverse Integer/Solution.java
class Solution {
    public int reverse(int x) {
        int r = 0, n = 1;
        if (x < 0) {
            n = -1;
            x = -x;
        }
        while (x > 0) {
            int t = x - (x / 10 * 10);
            r += t;
            if (x / 10 != 0) {
                if (r * 10 / 10 != r)
                    return 0;
                r *= 10;
            }
            x /= 10;
        }
        return r * n;
    }
}

File: 2025/7. Reverse Integer/solution.py
class Solution:
    def reverse(self, x: int) -> int:
        sign = -1 if x < 0 else 1
        x, y = abs(x), 0
        while x:
            x, y = x // 10,  y * 10 + x - (x // 10 * 10)
        return sign * y if y <= 2**31 else 0